### PR TITLE
MM-56391: fixed search bar moving up in ios

### PR DESCRIPTION
**Pull request by juliovillalvazo**
#### Summary
This PR solves the search bar staying on top when clearing the search input. What I did was add the Keyboard dismiss behavior when clearing the input therefore solving the issue.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/25843

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [X] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: Pixel 6 Pro Android Tiramisu

#### Screenshots

https://github.com/mattermost/mattermost-mobile/assets/51309531/cf40b202-0ef5-442b-b728-87fd3479c92f

#### Release Note
```release-note
Fixed search bar staying on top after clearing the input in the Search Screen.
```

### CODE REVIEW

File: package-lock.json
Comment (enahum) [inline package-lock.json:1]: let's revert the changes in this file

**Review by enahum — APPROVED** [review]
LGTM after reverting the changes in package-lock.json

File: package-lock.json
Comment (juliovillalvazo) [reply package-lock.json:1]: Hi @enahum I've solved conflicts in that file 😄

File: app/components/navigation_header/index.tsx [hunk 1]
```diff
@@ -62,13 +62,12 @@ const NavigationHeader = forwardRef<SearchRef, Props>(({
...
     const containerHeight = useAnimatedStyle(() => {
-        const minHeight = defaultHeight;
         const value = -(scrollValue?.value || 0);
         const calculatedHeight = (isLargeTitle ? largeHeight : defaultHeight) + value;
         const height = lockValue?.value ? lockValue.value : calculatedHeight;
         return {
-            height: Math.max(height, minHeight),
-            minHeight,
+            height: Math.max(height, defaultHeight),
+            minHeight: defaultHeight,
```
Comment (larkox) [inline app/components/navigation_header/index.tsx:70]: I don't see how this change makes things any different. Can you explain?

File: app/screens/home/search/search.tsx [hunk 2]
```diff
@@ -149,6 +149,7 @@ const SearchScreen = ({teamId, teams}: Props) => {
 
     const handleClearSearch = useCallback(() => {
         clearRef.current = true;
+        Keyboard.dismiss();
```
Comment (larkox) [inline app/screens/home/search/search.tsx:152]: I see this may be important, but I don't see how it handles the original issue.

File: package-lock.json
Comment (larkox) [reply package-lock.json:1]: Please revert these changes.

**Review by larkox — CHANGES_REQUESTED** [review]
I have a few non-blocking questions, but please, revert the changes in `package-lock.json`.

File: app/components/navigation_header/index.tsx (hunk 1 above)
Comment (juliovillalvazo) [reply app/components/navigation_header/index.tsx:70]: Hi @larkox, this change was not related to the issue, it was only to simplify code.

File: app/screens/home/search/search.tsx (hunk 2 above)
Comment (juliovillalvazo) [reply app/screens/home/search/search.tsx:152]: This is the line that solves the issue, if the keyboard was not dismissed the styles rendering the header were calculated as if the keyboard was still opened when the search was cleared causing the bug.

File: package-lock.json
Comment (juliovillalvazo) [reply package-lock.json:1]: I believe I have now reverted the changes, please let me know if I'm doing something wrong 🙏🏽

**Review by yasserfaraazkhan — CHANGES_REQUESTED** [review]
@juliovillalvazo can you look into Ci failure ? due to eslint errors.

**Comment by larkox** [comment]
@juliovillalvazo It seems something went weird with the merge / rebase. Can you take a look?

**Comment by juliovillalvazo** [comment]
> @juliovillalvazo It seems something went weird with the merge / rebase. Can you take a look?
Hi @larkox 
I believe with the changes I made the merge / rebase will now work, if not please let me know! 😄

**Review by larkox — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]
tested on android. after clearing the input box, the keyboard collapses as expected.
